src/core/utils/fileutils.cpp

```diff
@@ -174,6 +174,38 @@ QByteArray FileUtils::fileChecksum( const QString &fileName, const QCryptographi
   return QByteArray();
 }
 
+QString FileUtils::fileEtag( const QString &fileName, int partSize )
+{
+  QFile f( fileName );
+  if ( !f.open( QFile::ReadOnly ) )
+    return QString();
+
+  const qint64 fileSize = f.size();
+  QCryptographicHash hash( QCryptographicHash::Md5 );
+  if ( fileSize <= partSize )
+  {
+    if ( hash.addData( &f ) )
+    {
+      return hash.result().toHex();
+    }
+  }
+  else
+  {
+    QByteArray md5SumsData;
+    qint64 readSize = 0;
+    while ( readSize < fileSize )
+    {
+      hash.addData( f.read( partSize ) );
+      md5SumsData += hash.result();
+      hash.reset();
+      readSize += partSize;
```

Comment:
```suggestion
      readSize += 1;
```

Reply:
Discard my previous comment, it was with the wrong assumption that `int/int=float` in c++.